cpp/dolfinx/mesh/utils.h

```diff
@@ -153,4 +153,12 @@ compute_incident_entities(const Mesh& mesh,
                           const xtl::span<const std::int32_t>& entities, int d0,
                           int d1);
 
+/// Update mesh with a new set of ghosts.
+// Ghost can be removed or added according to the new ghost list.
+/// @param[in] mesh The mesh
+/// @param[in] dest Destination rank for mesh cells owned by this rank
+/// @return Mesh with new ghosts
+mesh::Mesh update_ghosts(const mesh::Mesh& mesh,
```

Comment:
This function name doesn't see right to me. It's not updating a mesh, but creating a new one.

I'm also not sure why one would use the function. Should it really work on a Topology object and return a new Topology object that can then me used to build a Mesh?

Reply:
At first I thought about modifying the mesh object, but the topology  has some states (connectivities, entities) that would be invalidated.  
I'll modify (simplify) the function to return a new topology, which could be used later for constructing a new mesh.